src/main/kotlin/no/nav/familie/ef/sak/vilkår/PatchVilkårController.kt

```diff
@@ -12,43 +13,61 @@ import org.springframework.web.bind.annotation.PostMapping
-                                                          type = VilkårType.DOKUMENTASJON_TILSYNSUTGIFTER,
-                                                          delvilkårsvurdering = DelvilkårsvurderingWrapper(delvilkårsvurderinger = listOf(
-                                                                  Delvilkårsvurdering(resultat = Vilkårsresultat.OPPFYLT,
-                                                                                      vurderinger = listOf(Vurdering(regelId = RegelId.HAR_DOKUMENTERTE_TILSYNSUTGIFTER,
-                                                                                                                     svar = SvarId.JA,
-                                                                                                                     begrunnelse = ""))))))
-                if (liveRun.skalPersistere) {
-                    secureLogger.info("persistererer vilkår for dokumentasjon av tilsynsutgifter på behandling med id=${it.id}")
-                    vilkårsvurderingRepository.insert(nyVilkårsvurdering)
+                secureLogger.info("bygger opp nytt delvilkår for om inntekt samsvarer med OS på behandling med id=${it.id}")
+                val gammelVilkårsvurderingEllerNull = eksisterendeVilkår.find { vilkår -> vilkår.type == VilkårType.INNTEKT }
+
+                if (gammelVilkårsvurderingEllerNull != null) {
+                    val nyDelvilkårsvurdering = lagNyDelvilkårsvurdering(it.id)
+                    if (liveRun.skalPersistere) {
+                        secureLogger.info("persistererer delvilkår for om inntekt samsvarer med OS på behandling med id=${it.id}")
+                        vilkårsvurderingRepository.update(gammelVilkårsvurderingEllerNull.copy(delvilkårsvurdering = DelvilkårsvurderingWrapper(
+                                delvilkårsvurderinger = gammelVilkårsvurderingEllerNull.delvilkårsvurdering.delvilkårsvurderinger.plusElement(
+                                        nyDelvilkårsvurdering))))
+                    }
+                } else {
+                    secureLogger.info("persisterer ikke delvilkår - fantes ingen vilkår for inntekt på behandling med id=${it.id}")
                 }
             }
         }
-        return Ressurs.success("patching av dokumentasjonTilsynsutgifter ble kjørt")
+        return if (liveRun.skalPersistere) Ressurs.success("patching av inntektsvilkår ble kjørt") else Ressurs.success("patching av inntektsvilkår ble ikke kjørt")
+    }
+
+    private fun lagNyDelvilkårsvurdering(id: UUID): Delvilkårsvurdering {
+        val fagsak = fagsakService.hentFagsakForBehandling(id)
+        val person = fagsakPersonService.hentPerson(fagsak.fagsakPersonId)
+        val fagsaker = fagsakService.finnFagsakerForFagsakPersonId(person.id)
+        val svar = if (fagsaker.overgangsstønad != null) SvarId.JA else SvarId.BRUKER_MOTTAR_IKKE_OVERGANGSSTØNAD
```

Comment:
Er denne diskutert med fag?

Kan kanskje være en idé å hente ut siste iverksatte behandling:

`behandlingRepository.finnSisteIverksatteBehandling(fagsakId)` 

Hvis denne eksisterer så kan vi svare "Ja", ellers "Bruker mottar ikke overgangsstønad" ?

Reply:
Hvis vi vil sjekke løpende så kan vi se på `andelTilkjentYtelse` fra svaret ☝️  og se om til-og-med-dato er etter dagens dato.

Reply:
Her fant vi ut etter nøye testing at "ikke_tatt_stilling_til" egentlig ikke er en støttet verdi for delvilkår. Med andre ord gikk vi for "oppfylt".